src/replay/replay_recorder.cpp

```diff
@@ -231,28 +332,53 @@ void ReplayRecorder::save()
                 StringUtils::xmlEncode(kart->getController()->getName()).c_str());
     }
 
+    // Calculate the unique identifier
+    // Part of it is random
+    // Part of it is based on race data
+    unsigned long long int unique_identifier = 0;
```

Comment:
What is the unique identifier needed for? Anyway, that should probably go to a new method dedicated to generating it. Also, this code is fairly weird, using a well-known hash function or UUIDs would be better if truly necessary

Reply:
Some unique identifier is needed for comparisons, as the temporary IDs are unstable (list sorts can shuffle files around, for example).

Calling classic hash functions/UUID is overkill for the use of this identifier (while part of the goal is that shared ghost files have different IDs, a random collision would not matter). Also, reversibility of the UID is a feature. Baking in some race data in it is useful for some consistency checks on a replay file. I'd even add more general data if I could have a 128-bits number rather than a 64-bits one.

Moving this to a separate function would certainly be better.